### DISCUSSION TITLE
Comments on Python package using RDKit shape and conformer generation for molecular size

### DISCUSSION THREAD

**Discussion by andrewtarzia** [rdkit/rdkit#4148] [Show and tell]
Hello all,

I have recently put together this package for calculating the molecular size (fits well with small molecule kinetic diameters) using the rdkit shape function: [https://github.com/andrewtarzia/mol-ellipsize](https://github.com/andrewtarzia/mol-ellipsize)

I am using it in a project (from my PhD, about 2 years late to publishing it.... ahah) that we hope to publish soon, but I was interested in feedback from you all about this approach. How does it compare to other approaches that people may have used? Is there a better way to implement the rdkit functionality I use?

Simply put:

- I use rdkit to generate conformers (although the code allows for and gives examples of arbitrary conformer input)
- I use rdkit to calculate the 'vdw cloud' of a molecule on a grid (note that since version 2019.02, the vdw scales that work best to reproduce kinetic diameters is 0.9).
- I find the minimum volume ellipsoid that fits those grid points, which defines 3 size axes

This process is fully automated from SMILES string to ellipsoid (hence, size) and is potentially useful in screening approaches.

**Comment by greglandrum** [comment]
Hi @andrewtarzia , thanks for letting us know about your work.
If you haven't done so already, I'd recommend comparing your new descriptor to some of the 3D descriptors which are already built into the RDKit: https://www.rdkit.org/docs/GettingStartedInPython.html#list-of-available-3d-descriptors

Doing something like generating values for a large collection of molecules/conformers and looking at the Spearman's correlation coefficient between the results of your calculation and the other single valued descriptors would be useful. I've been working on an RDKit blog post on the topic of comparing 3D descriptors... it's not done yet, but this early partial draft demonstrates how I've been thinking about the problem: https://github.com/greglandrum/rdkit_blog/blob/master/notebooks/Comparing%203D%20Descriptors.ipynb

**Reply by andrewtarzia** [reply]
Hey @greglandrum , I am running a script for doing this now.

The measure that I implement is more costly than any of those descriptors, but we have found good agreement with the kinetic diameters of sets of small molecules. I am writing this into a JOSS paper, and so improvements are always appreciated!

**Reply by andrewtarzia** [reply]
Found an issue, potentially, where CalcSpherocityIndex (unlike the other 3D descriptors) does not give conformed dependant values using rdkit version 2019.09.2.0 and :

```python
def calculate_rdkit_descriptors(molecule, cid):
    descriptor_functions = desc_list()

    return {
        i.__name__.replace('Calc', ''): i(molecule, confId=cid)
        for i in descriptor_functions
    }

````

**Reply by greglandrum** [reply]
Yep, that's a bug. There's some overly agressive caching of computed values happening.
A workaround is to call `molecule.ClearComputedValues()` at the beginning of `calculate_rdkit_descriptors()`.

**Reply by andrewtarzia** [reply]
Thank you, that is handy. It is taking my laptop some time to generate all these structures, so I might stop at a subset of the platinum database, and ignore sphericity index (as its not a measure of size, that also makes some sense). I will upload some results in the coming days.

**Reply by andrewtarzia** [reply]
Hey @greglandrum ,

I have done a fairly simple comparison that I think is most relevant (for 700 molecules in the platinum dataset). I have compared the RadiusofGyration from rdkit for each conformer vs. my measure of molecular size (see the bottom for a plot). My size measure is parameterized to match defined kinetic diameters, which are physically similar to R_g in that they represent a spherical average of the molecule. However, the defined kinetic diameters are for smaller, less flexible molecules, so the spherical assumptions hold better. The data shows, which is not surprising, that these two measures do not correlate well in most cases. This is not surprising because as you have more flexible molecules, the less spherical they become and then the more relevant the ellipsoidal shape is to measuring size. I guess this means that this measure has its place for certain problems as existing measures will not capture this information? However, as the plot at the bottom shows, the overall relationship between the two size measures is ok, its just within sets of conformers that flexibility is important? Maybe that conclusion is illogical...

I hope this helps and I would be happy to help out with the blog post if you think it is relevant. The code to run the comparisons and extract the coefficients (as well as the data!) has been pushed to the github for the code: https://github.com/andrewtarzia/mol-ellipsize/tree/main/examples

A plot of the RG vs the molecular size from my code is attached here for all conformers of all molecules I tested. The general trend is not actually that bad.

[rg_vs_mid.pdf](https://github.com/rdkit/rdkit/files/6532168/rg_vs_mid.pdf)
